dataframely/columns/binary.py

```diff
@@ -0,0 +1,59 @@
+# Copyright (c) QuantCo 2025-2025
+# SPDX-License-Identifier: BSD-3-Clause
+
+from __future__ import annotations
+
+from typing import Any
+
+import polars as pl
+
+from dataframely._compat import pa, sa
+from dataframely.random import Generator
+
+from ._base import Check, Column
+from ._registry import register
+
+
+class _BaseBinary(Column):
+    def __init__(
+        self,
+        *,
+        nullable: bool | None = None,
+        primary_key: bool = False,
+        check: Check | None = None,
+        alias: str | None = None,
+        metadata: dict[str, Any] | None = None,
+    ):
+        super().__init__(
+            nullable=nullable,
+            primary_key=primary_key,
+            check=check,
+            alias=alias,
+            metadata=metadata,
+        )
+
+    @property
+    def dtype(self) -> pl.DataType:
+        return pl.Binary()
+
+    def sqlalchemy_dtype(self, dialect: sa.Dialect) -> sa.sql.sqltypes.LargeBinary:
+        if dialect.name == "mssql":
+            return sa.VARBINARY("max")
+        return sa.LargeBinary()
+
+    @property
+    def pyarrow_dtype(self) -> pa.DataType:
+        return pa.large_binary()
```

Comment:
Pay close attention to these lines. I'm not familiar enough with sqlalchemy or pyarrow to know what the right values should be here